File: Inheritance/hierachicalinheritance.py
class person:
    def __init__(self,name,age):
        self.name=name
        self.age=age

    def showdata(self):
        print("name=",self.name)
        print("age=",self.age)

class teacher(person):
    def __init__(self,name,age,post):
        super(name,age)
        self.post=post

    def showdata(self):    Inteegr.parseInt(JOptionPane.showInputDialog())
            super().showdata()
            print("post",self.post)

class student(person):
     def __init__(self,name,age,standard):
          super(name,age)
          self.standard=standard

     def shwodata(self):
         super().showdata()
         print("standatd=",self.standard)
obj=teacher("manish",13,"TGT")
obj1=student("buddy",18,"ty")
obj.showdata()
obj1.showdata1()


# In python The Hierachical Inheritance is not support
    
        

File: Hierachicalinheritance.java
class Person
{
	private String name;
	private int age;
	
	public Person(String s,int a)
	{
		name=s;
		age=a;
	}

	public void display()
	{
		System.out.println("name:"+name);
		System.out.println("age:"+age);
	}
}
class Teacher extends Person
{
	private String post;
	
	public Teacher(String s,int a,String c)
	{
		super(s,a);
		post=c;
	}
	public void display()
	{
		super.display();
		System.out.println("post:"+post);
	}
}
class Student extends Person
{
	private int standard;
	public Student(String s,int a,int st)
	{
		super(s,a);
		standard=st;
	}
	public void display()
	{
		super.display();
		System.out.println("Standard:"+standard);
	}
}

public class Hierachicalinheritance
{
	public static void main(String[] args)
	{
		Teacher t=new Teacher("manish",23,"TGT");
		Student s=new Student("madhu",56,10);
		t.display();
		s.display();
	}
}